cdap-data-fabric/src/main/java/co/cask/cdap/data2/sql/PostgresSqlStructuredTable.java

```diff
@@ -138,6 +130,93 @@ public void upsert(Collection<Field<?>> fields) throws InvalidFieldException, IO
     }
   }
 
+  @Override
+  public boolean compareAndSwap(Collection<Field<?>> keys, Field<?> oldValue, Field<?> newValue)
+    throws InvalidFieldException, IOException {
+    LOG.trace("Table {}: CompareAndSwap with keys {}, oldValue {}, newValue {}", tableSchema.getTableId(), keys,
+              oldValue, newValue);
+    tableSchema.validatePrimaryKeys(keys.stream().map(Field::getName).collect(Collectors.toList()), false);
```

Comment:
I feel that it would be better to have a method tableSchema.validatePrimaryKeys(Collection<Field<?>>). Applying streams here looks a little like overkill.